### ISSUE TITLE
docs: "definitive" organization

### ISSUE THREAD

**Issue by dmpetrov** [iterative/dvc.org#144]
UPDATE: Jump to https://github.com/iterative/dvc.org/issues/144#issuecomment-841960760; E: https://github.com/iterative/dvc.org/issues/144#issuecomment-1206648437

---

We need to add an additional level to the user guide:
1. Basic
- DVC Files and Directories
- DVC File Format
- External Dependencies
- External Outputs
- Update a Tracked File
- Anonymized Usage Analytics
2. Customization
- DVC Shell Autocomplete
- IDE Plugins & Syntax Highlighting
- Development Version
- Contributing

---

UPDATE: See discussion about structure below, and more subtasks in https://github.com/iterative/dvc.org/issues/144#issuecomment-584441916 further below

**Comment by Dhiraj240** [comment]
@dmpetrov I am looking to contribute to the project ```Improving and expanding User Guide``` in this upcoming Google Season of Docs 2019. I came here just after the organisations were announced, I already have experience with the opensource organisations. You can see my GitHub profile and Gitlab profile: https://gitlab.com/Dhiraj240?nav_source=navbar
May I know on how to get started with this project so that I will be able to submit a proposal during the proposal period. is there any gitter channel for communication?

**Comment by dmpetrov** [comment]
@Dhiraj240 thank you for your interest! Yeah, Discord is a better place to discuss. I'm seeing you have already started.

**Comment by Dhiraj240** [comment]
First, I will work on some issues by this week then after that please guide me. On a weekly basis, I will solve issues. :smile:

**Comment by dmpetrov** [comment]
Awesome!

**Comment by jorgeorpinel** [comment]
The user guide structure has changed quite a bit since this was opened. Is it still desired?

**Comment by shcheklein** [comment]
Yes, I think it still relevant. We still don't have a good intermediate structure for the UG. Not sure if it overlaps with some other tickets or duplicates them.

**Comment by jorgeorpinel** [comment]
OK. I'd just note that to avoid 4 levels (excessive clicking) we'll need to remove the **Contributing** submenu and just list both Contributing guides inside **Customization** directly.

**Comment by shcheklein** [comment]
@jorgeorpinel don't consider that split in the ticket description as a final one. It's just an example to illustrate the idea.

**Comment by jorgeorpinel** [comment]
- [ ] We should also move the Google API Privacy Policy to another location (not really a user guide) and even consider having it as a link without nav entry (implementing #731).

**Comment by VANRao-Stack** [comment]
> Other general doc structure changes:
> 
> 
>   
>     
>    #425 Absorb Understanding DVC into existing sections
> 
>   
>     
>    user-guide: extract part of config cmd ref into the user guide #340
> 
>   
>     
>    #1035 merge doc parts into super sections?
> 
>   
>     
>    check #144 (comment) above
> 
>   
>     
>    We should also move the Google API Privacy Policy (https://dvc.org/doc/user-guide/privacy) to another location (not really a user guide) and even consider having it as a link without nav entry (implementing #731).
> 
> 
> 
> 
>   
>     
>    Some of these movements will require redirects, at least for some time.

I have created a pull request for absorbing the understanding DVC section. Please take a look at it and let me know if there are any changes that I may need to bring about.

**Comment by jorgeorpinel** [comment]
@VANRao-Stack thanks, I will check it out. You are also involved with https://github.com/iterative/dvc.org/issues/614#issuecomment-630561304 though, please pick one to focus on for now and let us konw. Thanks

**Comment by jorgeorpinel** [comment]
@VANRao-Stack I know this issue is marked as `good-first-issue` and thanks for taking the initiative but I think this is kind of an epic with many subtasks:

- The general reorg of User Guide as described in the description
- The other related tasks listed in https://github.com/iterative/dvc.org/issues/144#issuecomment-584441916

Your PR is a good start but please focus on a single subtask to make this more manageable.

**Comment by jorgeorpinel** [comment]
> > The user guide structure has changed quite a bit since this was opened. Is it still desired?
>
> Yes, I think it still relevant. We still don't have a good intermediate structure for the UG.

So is the intention of this ticket to figure out the most efficient way to organize our current and future docs in a sustainable way? A kind of "ultimate solution" to docs structure? (If so I can update the issue's title and desc.)

I think for that we would need to analyze the website traffic, search results, conversions, etc. to make sure the stuff we put close to the surface is the most needed, and to determine which things can get buried or even hidden (no nav entry), even deleted (left for blog posts and support channels to cover).

**Comment by jorgeorpinel** [comment]
Here's a proposal:

- /docs 🏠 (Home) with linked list to **Get Started** topics
  - cases 💡 Use Cases  _- may extract to /cases (outside of docs) later_
  - guide 📖 User Guide: main container (separates from refs.) _- consider skipping this level_
    - what-is-dvc  _- maybe merge with / home_
    - install/  Per OS, Pre-release, Shell completion, Env Tips (IDEs, Win)
    - dvc-files/  **Metafile Formats and Internals** (.dvc/) _- maybe reorg into topics below_
    - data-mgmt/ 🗃️ **Data Management** `topic`
      - start-ver  **Get Started**: Data & Model Versioning
      - start-access  **Get Started**: Data Access
      - dvc-cache/  Cache structure, Run cache, Shared cache
      - optimization  Optimization & Link types
      - external-data  #520
      - dvc-remotes/  Config refs + setup guides per remote type
    - data-pipelines/ 🔃 **Data Pipelines** `topic`
      - start-dag  **Get Started**: Data Pipelines (DAG)
      - start-metrics-params  **Get Started**: Parameters, Metrics/Plots
      - stages
        - foreach
      - dependencies
        - parameters
      - outputs
        - metrics-plots
      - templating (parametric dvc.yaml)
    - exp-mgmt 👩‍🔬 **Experiment Management** `topic`
      - start-experiments  **Get Started**: Experiments
      - exp-workflow
      - checkpoints
    - how-to/  **How Tos**
    - contributing/
    - troubleshooting
  - cli-ref 👨‍💻 Command Reference
  - api-ref 🐍 Python API Reference

I'm not quite sure about putting Data Pipelines inside Experiment Management or whether it needs it's own top section in the UG. They seem to overlap a lot but we could start by separating them instead

Cc @shcheklein @dberenbaum @casperdcl @iesahin

**Comment by casperdcl** [comment]
Re: doc-vs-docs, a slight update to my "slight preference for fewer chars" https://github.com/iterative/dvc.org/issues/2443#issuecomment-832808614:

Mathematics -> maths (UK)/math (US). Documentation -> doc (universal), surely? Please don't tell me it's doc (UK)/docs (US) :confused:

**Comment by casperdcl** [comment]
I wouldn't nest everything under a "guide" (or whatever name) level - seems like unnecessary user navigation difficulty.

**Comment by dmpetrov** [comment]
> data-mgmt/ 🗃️ **Data Management** `topic`

Is **Data Management** a good name for the topic? DVC versions and transfers data that goes to ML model. Data management seems like a broader term that might include data sources before it goes to DVC - Data Wherehouse/DB or a directory with "immutable" data / S3. The only scenario when DVC does proper data management is [data registry](https://dvc.org/doc/use-cases/data-registries). I don't see any better name so far 😬 Do you have any ideas?


> data-pipelines/ 🔃 **Data Pipelines** `topic`

**Data Pipeline** refers to data engineering tools such as AirFlow. I'd suggest using **ML Pipelines** that might be also not the best but seems slightly better.

We probably need an additional section on **model management** that should include all the metrics and plot navigation commands - `dvc metrics/plots`. These commands are pretty independent (even should work without `dvc.yaml` is a terget is specified) and make Git repo metrics-driven. Also, extracting model management into a separate topic will reduce the complexity of ML-pipeline and experiment topics.

> exp-mgmt 👩‍🔬 **Experiment Management** `topic`

**Experiment Management** - "Experiment Tracking" is another term but I'm not sure which one is beter.

**Comment by iesahin** [comment]
* IMHO it may be `doc` or `docs`, but if it could be identical with the file path (currently, `docs`), I could navigate the [links] with my `gf`.

* For _data pipelines_, another alternative may be _model pipelines_, as it's usually the end product, or _data-model pipelines._ (Data in, model out.)

**Comment by jorgeorpinel** [comment]
> Documentation -> doc (universal), surely?

@casperdcl it's not universal but also I don't thin it's related to US vs UK. TBH I wasn't proposing to change it, it just came out that way. But a quick check gives me the impression that `docs` is more common e.g. terraform.io/docs, docs.aws.amazon.com, developers.google.com/docs, docs.microsoft.com/en-us/azure

> wouldn't nest everything under a "guide" (or whatever name) level

I incline the same way but you would need to scroll a lot to find the references without that level so I'm not sure.

@dmpetrov agree to rethink the topic titles. But I think that if we can agree on the grouping of content, we can do that during the PR(s).

> additional section on model management that should include all the metrics and plot navigation commands - dvc metrics/plots. These commands are pretty independent ... and make Git repo metrics-driven

Maybe (structure-wise):

- Handling Data (Sets)
- Modeling Pipelines
- ML Model Optimization
- Experiment Tracking

@iesahin `gf`?

**Comment by casperdcl** [comment]
I assume he meant `gfm` (*github flavoured markdown*) rather than the usual abbreviation for *girlfriend*

**Comment by shcheklein** [comment]
(I assume that doc/docs, `cases`, etc - are not about changing names and URLs, but rather local names here to discuss the structure and refer to the sections faster)

- I don't see get started here. I see it's hidden under UG, but I would not do this. It's better to have it as a top-level section. `cases` are too abstract, it better to start with usual things about the product - how to start, how to install, etc. Happy path pretty much.
- Install - we had a quite long discussion and decided that it's good to keep it top level, right?
- Guide should be starting with some overview - basic concepts. Then go into details by specific topics. I agree with @dmpetrov re the names. We should iterate on this. Even though I can see clear data management scenarios besides data registry (e.g. shared cache).
- Agreed on including Model Management that expands on plots/metrics/etc. And this way we won't have to squeeze this into the Data Management by doing somethings like "Data and Model ...."

--------

Overall, this suggestion has a lot of name changes, structural changes that are hard for me to justify. I would start by cleaning this up step by step and by introducing sections in the UG as we go (e.g. when we move shared cache). We'll have more clarity after that and generalize things.

**Comment by iesahin** [comment]
`gf` is _go to file_ in vim. 😁 It's easier to navigate the links by having the cursor on link and type `gf`. This feature is also available in VS Code markdown plugins I think. We can't navigate the links offline because links are `/doc/` but paths are `docs/`. 

I also think it's a not a big deal. I have other means to navigate.

**Comment by jorgeorpinel** [comment]
> get started here. I see it's hidden under UG, but I would not do this. It's better to have it as a top-level section. 

@shcheklein a) if we remove the UG level (https://github.com/iterative/dvc.org/issues/144#issuecomment-841990301) then they're the first pages under each topic, b) we could repeat those pages in 2 places, c) we can list/link them all directly in /docs home.

I also don't mind keeping a top-level Gs group for now but its structure looks a lot like the proposed top-level (or UG) structure: Data, Pipelines, Models, Experiments, which may make the navigation confusing.
Also I expect it will grow even longer and it's starting to look like a full tutorial... But that's a separate issue.

> cases are too abstract, it better to start with usual things about the product

I moved them up thinking that we won't even keep them under /docs. But as long as they're in here sure, we can keep them after GS

> Install - we had a quite long discussion and decided that it's good to keep it top level, right?

It's in the top if we remove the UG level (UG is an abstract docs container in the proposal).

> Guide should be starting with some overview - basic concepts

True, I forgot about Basic Concepts but it doesn't exist yet (there's an issue for that).

> this suggestion has a lot of name changes, structural changes that are hard for me to justify.

Really the main proposal would be to eliminate the UG level and regroup most guides into 4 topics instead. The other big change was the redistribution of Get Started entries but it's not needed now. In summary:

/docs Home
Install
Get Started
Data Management
Modeling Pipelines
ML Model Optimization
Experiment Tracking
Cmd Ref
API Ref
Misc?

**Comment by casperdcl** [comment]
wrt https://github.com/iterative/dvc.org/issues/144#issuecomment-841960760:

- `/doc/cases`

> may extract to `/cases` (outside of docs) later rel #820

I'd say merge with `/features` and `/`. Complex tutorial-like bits to be extracted to other doc pages.

- `/doc/guide` -> `/doc`

> guide book **User Guide** (main container to separate from refs.) _- could even remove this 
level?_

`/guide/` is a meaningless level - apparently only exists to collapse irrelevant info for those looking for `/doc/api-ref/` and `/doc/cli-ref/`. There are other ways to emphasise `*-ref` (e.g. italics, bold, etc). Note that https://docs.docker.com has separate roots for "Guides," "Product manuals," "Reference," and "Samples."

- `/doc/dvc-files/`

> **Metafile Formats and Internals** (.dvc/) _- maybe reorg into topics below_

This should appear next to CLI-ref and API-ref because it's at that level of detail. Also should be called "DVC file formats/project directory structure" or something more descriptive.

- `/doc/data-mgmt/` `/doc/data-pipelines/` `/doc/exp-mgmt/`

I agree with the other comments that these are misleading names.

Also presumably these will be slightly more specific that `/cases`/`/features`/`/` but explicitly NOT tutorials.

- `/doc/how-to/`

There are presumably tutorials.

We desperately need to have a list of things which we consider synonymns because otherwise the language barrier seems to be the biggest problem when communicating with each other.

- `/doc/contributing/`

Doesn't really have much business being here, I think. Should be in repo `CONTRIBUTING.md`/`.github/CONTRIBUTING.md` or repo's [wiki](https://github.com/iterative/dvc.org/wiki).

- `/doc/troubleshooting/`

This should be at `/help` or `/support` surely?

**Comment by jorgeorpinel** [comment]
@casperdcl thanks. I think we can worry about moving /cases out of docs and about outliers (contrib, troubleshooting) in a 2nd iteration.

Yes, my main proposal is to remove the /guide level. I don't know if it's completely meaningless (the word "guide" gives you a good idea of what you'll find inside) but it's better to reorg into the topics discussed above. And I think we can find the right names as we work on that change, once/if we all agree on this.

The Q on whether /doc/dvc-files is a reference or a guide is also not so simple so I'd let it be for now (same as for where to list /start pages).

**So to recap once more**: a first reorg iteration would basically split the Guide into 4 topics. All in favor say aye ✋

**Comment by casperdcl** [comment]
> my main proposal is to remove the /guide level. I don't know if it's completely meaningless

I agree, this is what I meant. More accurately I mean "it's not useful to nest things under a level whatever you call it."

**Comment by shcheklein** [comment]
> Yes, my main proposal is to remove the /guide level

Please, let's not do this. A few reasons from the top of my head:

- we'll be mixing apples and oranges in the top level structure - high level doc-related components like Home, Cmd Ref, etc and actual DVC specific items. The idea behind the existing structure is to reflect 3-4 major parts of _any_ docs - Refs, Guides, Quick Start, etc.
- If we do this - it won't scale well, we'll be adding more stuff there. Where will we put Contrib? Troubleshooting? How to, etc? We'll completely hide other _major_ parts - Cmd Ref, Get started, etc
- not the right time for this? Let's first try to properly consolidate things under Guide and then we can decide?

**Comment by shcheklein** [comment]
> /guide/ is a meaningless level - apparently only exists to collapse irrelevant info for those looking for

I don't agree. We have never spent enough time to do it right - that's why it looks this way. If we move stuff out, then all docs look like a mess. If we plan on how do we move and instead reorganize inside - it'll look better.

**Comment by shcheklein** [comment]
`How tos` are technically tutorials, but they have a very precise angle - how to solve a very niche problem, vs general tutorials. You see the difference by their titles.

**Comment by jorgeorpinel** [comment]
> Let's first try to properly consolidate things under Guide and then we can decide?

OK. So would reorganizing most guides into 4 topics as (sub-levels of /doc/guide) be a good first step? Basic Concepts can stay in the beginning of the Guide section (before the 4 topics). How To and Outlier pages can be after (I think those are found most often from searches or links from other docs). 

```
...
Guide
  DVC Concepts ...
  Data Management (1) ...
  Modeling Pipelines (2) ...
  ML Model Optimization (3) ...
  Experiment Tracking (4) ...
  How to ...
  Troubleshooting
  Contributing
```
\* Names aren't final

Cc @casperdcl ☝️

**Comment by iesahin** [comment]
I don't have strong opinions on this. I'm mostly a search guy and never had difficulty to navigate in the docs. :)

**Comment by jorgeorpinel** [comment]
Right. It's not just about the nav though. This organization will help us compartmentalize the areas of docs better. It should help with planning and knowing we are covering enough types of docs for enough types of users.

**Comment by casperdcl** [comment]
> The idea behind the existing structure is to reflect 3-4 major parts of any docs - Refs, Guides, Quick Start, etc.

Hmm I'm thinking about a traditional book, report, article, journal etc:

1. cruft
2. main body
3. cruft

AKA:

1. intro, preface, acknowledgements, installation
2. main body
3. conclusion, bibliography, glossary, index, command reference

Where the unique thing about the "main body" is that (unlike all the other parts) is it's never actually labelled. You don't have a chapter titled "main body" - at most you use different numbering to make a distinction, but not a nested level, e.g:

```
i. Use Cases
ii. Installation
1. Data Registry
2. ML Pipelines
3. ...
A. Command Ref
B. API Ref
```

So basically I'm not accustomed to the idea of a main body (in this case labelled guide) nesting.

**Comment by shcheklein** [comment]
Interesting analogy! To me docs != book (for a lot of reasons). But if you prefer that analogy - User Guide == DVC Book for me and should be structured the way you described. Other sections = marketing materials, quick brochures around, etc.

**Comment by casperdcl** [comment]
:shrug:

**Comment by jorgeorpinel** [comment]
From chat with @dmitry. Some feedback from advanced DVC users on docs experience:

Docs now are too command-centric; people can sometimes get lost in the details; unclear where to find the information they need (or what they even need in the first place).

> p.s. @jendefig also mentioned this. More detailed notes [here](https://www.notion.so/iterative/XP-Inc-d6550d24753049bfb1b04fa2b8b7f3c8) (internal)

Idea on the structure of **experience** (not an exact map to sitemap necessarily):

0. Landing on home page isn't showing a clear **BIG picture** of where in the ML process DVC fits (the gap)
  > DVC website doesn't have to be only about DVC -- it's about a workflow
1. High-level **scenarios** (use cases?) e.g. workflow for ML exps; sxp mgmt; versioning; GitOps; etc.
  > We can do a better job of referring to other products (MLEM, CML, etc.) at this level
2. How do we teach the tools? Quick Start: single introductory section to grasp all the scenarios (features) vs. Get Started tutorials that are still introductory but go into more detail (take more time)
3. Usage Guides & References
4. Everything else? (How-tos, Troubleshooting, etc.)

**Comment by iesahin** [comment]
Repeating the questions in https://www.notion.so/iterative/wip-Audiences-ccb7abb9a198476aa0c01581479c5eaf

* Who uses DVC?
* What do they use DVC for?
* How do they use DVC?

The answers to these questions should shed some light on the docs organization. We can't be everything to everyone, so we must select the *most obvious audience* and make docs organization easier for them.

**Comment by jendefig** [comment]
@iesahin @jorgeorpinel 
While I do agree that organizing docs should be a function of audience, I think the need and request for having workflows isn't an audience question, it's because as the tools have more features, the docs grow and become harder to absorb. (It's a lot to read/take-in).  We have to figure out a way to simplify the organization of the docs to seem less overwhelming.  All the documentation on commands must be there.  It's a question of ingestion and that ingestion experience.

> Landing on home page isn't showing a clear BIG picture of where in the ML process DVC fits (solution positioning)
I agree with this one hundred percent.  Need a big picture a-ha moment right from the start. And it needs to be a sharable image.

On 1. Are we sure that Use Case is the best term?  We should use the language they are coming to us with.  I'm curious what terminology was used in @dmpetrov's discussions with the same request.  It is interesting that "command-centric" was used in what we both heard, but in what they were looking for instead, what terminology did they use? (What makes sense to our audience?) (Maykon definitely used "workflow")

Also regarding this in particular.  Why don't we just crowd source the Community?  Ask the question:  "To better serve your needs in our docs, we are asking for you to fill in and upvote items in this document (public in Notion?) on what workflows you would like to see covered in our docs.  For example, finish the search "How do I _____ ?"  (or however we should ask this).

Also from talking to Community members, it seems they go through steps in their usage/adoption.  Maybe we need to change the buttons on the main page to these milestones?  I want to:
Version my Data
Set up my pipeline
Version my Models
Version my Experiments
Examine my Results
Automate testing

Regarding the "Big picture" image, it would be awesome to be able to click on which section of the process they are dealing with to get to the corresponding docs about our tools.  (Yes, I dream big ;) )


On 2. Quick start vs. get started?  This would be confusing.  Why not Features and Tutorials? with a search capability in each

Not sure about 3 and 4. need more rumination.

**Comment by jorgeorpinel** [comment]
The top priority should be to have a clear visual for the big picture covering the entire ML lifecycle and where DVC comes in.

Other than that, the larger challenge is achieving a docs structure that can fit a lot of content in a way that seems simple and clear. I'm not sure that can be done under a single tree like the navigation menu we have on the left side.

💡 Maybe we need an orthogonal separation by features (mapped to the big picture above) that essentially filter content depending on the focus of each user (similar to the proposed Get Started trails). There would be some overlap in terms of explanation and reference docs, but a single Quick Start one-pager in each one and/or a bit longer Learn "tutorial" (current G/S pages)

On the term "use cases" we can definitely change it to "scenarios", "solutions", "why DVC", or something else. They should probably get out of the Docs section and be proper [landing pages](https://unbounce.com/landing-page-articles/what-is-a-landing-page/) instead of the current /features (this has been discussed several times).

**Comment by iesahin** [comment]
What I had in mind talking about "audiences" is that the workflows and expectations of these audiences vary considerably. We don't have a "single basic workflow" as Git may have. DVC's MLOps workflow(s) are considerably different than Data Science workflow(s), and IMO the best way to model this is via answering the "who?" question. 

If we start from "who", then it's easier to think about "what do these users do in 80% of their time with DVC?" question that will lead us to specific workflows. 

On top of this, we can have several different ways to structure the documents, e.g., longer tutorials vs quick get started. I'd like to have some tests and feedback mechanisms to measure the merits of these before deciding.

**Comment by casperdcl** [comment]
I like this idea. A popup perhaps linking to relevant "getting started" sections?

> Maybe we need to change the buttons on the main page to these milestones? I want to:
> Version my Data
> Set up my pipeline
> Version my Models
> Version my Experiments
> Examine my Results
> Automate testing

**Comment by jorgeorpinel** [comment]
UPDATE!

I think that this issue is too broad so we may never be satisfied with a definitive plan for this.

Also, the main remaining problem seems to be in the User Guide (book-like explanation content) and we have several "smaller" epics to tackle that, specifically:

- https://github.com/iterative/dvc.org/issues/550
- https://github.com/iterative/dvc.org/issues/2856
- https://github.com/iterative/dvc.org/issues/2883
- https://github.com/iterative/dvc.org/issues/2911
- https://github.com/iterative/dvc.org/issues/72
- https://github.com/iterative/dvc.org/issues/899

So we should probably close this, as long as those issues reflect the directions we intend, as described in https://github.com/iterative/dvc.org/issues/144#issuecomment-1000481879 above:

> Docs now are too command-centric; people can sometimes get lost in the details; unclear where to find the information they need...
> 0. Landing on home page isn't showing a clear BIG picture; 
> 1. High-level scenarios (use cases?
> 2. How do we teach the tools? Quick Start
> 3. **Usage Guides (explanations and/or walkthroughs)** & References (purpose details, semantics)
> 4. Everything else? (**How-tos**, Troubleshooting, etc.)

The only other thing from ☝🏼  that's not covered in the UG epics mentioned before is the home page redesign, but I can create a separate issue for that. WDYT @dmpetrov @shcheklein @casperdcl ? Thanks

**Comment by casperdcl** [comment]
> I think that this issue is too broad so we may never be satisfied with a definitive plan for this.

to me we can definitely come up with some action points

- [ ] are all the scenarios from [the OP](https://github.com/iterative/dvc.org/issues/144#issue-393658581) covered?
- [ ] [this popup/design idea](https://github.com/iterative/dvc.org/issues/144#issuecomment-1012399657)
- [ ] design: draw more attention to search box
- [x] ~agree on layout~ (neverending, not as important as above)

**Comment by dberenbaum** [comment]
@jorgeorpinel and I discussed some of these issues. A summary of some of my takeaways:

* We need to prioritize between changes to:
  * Landing page: https://github.com/iterative/dvc.org/issues/3833.
  * Get started: need to discuss plans for "trails," but goal should be address high-level use cases.
  * User guide: we are already working on #3413 and #3830, so I think it would make sense to prioritize this.
* User guide plans:
  * Split into high-level feature explanations and how-to: each high-level feature can be a section, and how-to can be an additional section with small pages addressing more low-level scenarios,
  * Thoughts on existing pages:
    * What is DVC: dissolve into some page outside the user guide (and maybe in the home page/outside of docs).
    * Project structure: move this into some "reference" section ("Metafile Schema Reference"?) -- could break reference into CLI, API, and this.
    * How to: keep it at the bottom of the user guide with subpages.
    * Setup GDrive remote: put into data management section of user guide or make into blog post (I prefer the first if we can support it).
    * External Deps/Managing External Data: put into pipelines section of user guide or make them only reachable via links.
    * Running on Windows: put in how to or part of https://dvc.org/doc/install/windows.
    * Troubleshooting/Glossary: put at bottom of the user guide.
    * Related technologies: dissolve into some page outside the user guide (and maybe in the home page/outside of docs).
    * Anonymized usage analytics: put at bottom of the user guide or make it only reachable via links (I prefer to keep it somewhere so it doesn't look like we are hiding this info).

cc @shcheklein

**Comment by shcheklein** [comment]
Alternative that I had in Slack (just as an idea):

```
What is DVC (move related techs inside it)
Project Str
Experiments Mng (Visualzing Plots should be inside)
Data Management (Remote, Large Data, External Data)
Troubleshooting
How to (move Windows inside, may even Google Drive for now)
Telemetry Policy
Glossary (I would remove it - I don;’t think it;’s useful tbh - why do we need it)?
```

**Comment by jorgeorpinel** [comment]
> What is DVC (move related techs inside it)

Currently, I started disintegrating that one as we didn't see it fitting the goals of the user guide as stated above (high-level feature explanations).

However, 💡 we could turn it into an **Overview** instead, which would combine its existing [Core Features](https://dvc.org/doc/user-guide/what-is-dvc#core-features) section as well as [Related Technologies](https://dvc.org/doc/user-guide/related-technologies). That way we take care of 2 birds (from the bullet list above) with one stone 🐦 🐦 🪨

**Comment by dberenbaum** [comment]
#4006 looks like a nice move in the right direction!

> Alternative that I had in Slack (just as an idea):
> 
> ```
> What is DVC (move related techs inside it)
> Project Str
> Experiments Mng (Visualzing Plots should be inside)
> Data Management (Remote, Large Data, External Data)
> Troubleshooting
> How to (move Windows inside, may even Google Drive for now)
> Telemetry Policy
> Glossary (I would remove it - I don;’t think it;’s useful tbh - why do we need it)?
> ```

Maybe we need an additional `Pipelines Mgmt` section in addition to `Experiments Mgmt` and `Data Mgmt` (esp since we just created this section 😄 )?

**Comment by dberenbaum** [comment]
Hm, now I see that pipelines is complicating things since pipelines are so tightly coupled to data management.

That could make sense if we position data management as including the pipelines to generate the data 🤔 .

**Comment by jorgeorpinel** [comment]
> Maybe we need an additional Pipelines Mgmt section

We have https://dvc.org/doc/user-guide/pipelines and https://github.com/iterative/dvc.org/issues/2883.

> pipelines are so tightly coupled to data management

I think that for the get started is OK to cover pipelining basics inside Data Management but the feature set is so rich that it does deserve a user guide section. Se can still mention and link a lot about it from data mgmt pages.

**Comment by jorgeorpinel** [comment]
> Glossary (I would remove it - I don’t think it’s useful tbh - why do we need it)?

No strong opinion but I like it. It's at the bottom of the section so it's not in the way either.

p.s. probably the page I visit more from Git docs is https://git-scm.com/docs/gitglossary (but then again I work a lot with wording and definitions).

**Comment by shcheklein** [comment]
Yes, I think we are good to close it for now and iterate on specific tickets.

**Comment by dberenbaum** [comment]
The definitive organization is complete, never to be touched again 😄

**Comment by jorgeorpinel** [comment]
"never"
